amazonka-test: DeriveLift lift instances?

amazonka-test has a bunch of manual `Lift` instances which raise warnings about missing methods. `-XDeriveLift` has been in since GHC 8.0.1 and `-XStandaloneDeriving` since 6.8.1; should we just do that instead (except for `DiffTime`, where we don't have the constructor)?

```
src/Test/AWS/TH.hs:25:10: warning: [-Wmissing-methods]
    • No explicit implementation for
        ‘liftTyped’
    • In the instance declaration for ‘Lift (Time a)’
   |
25 | instance Lift (Time a) where
   |          ^^^^^^^^^^^^^

src/Test/AWS/TH.hs:28:10: warning: [-Wmissing-methods]
    • No explicit implementation for
        ‘liftTyped’
    • In the instance declaration for ‘Lift UTCTime’
   |
28 | instance Lift UTCTime where
   |          ^^^^^^^^^^^^

src/Test/AWS/TH.hs:34:10: warning: [-Wmissing-methods]
    • No explicit implementation for
        ‘liftTyped’
    • In the instance declaration for ‘Lift DiffTime’
   |
34 | instance Lift DiffTime where
   |          ^^^^^^^^^^^^^

src/Test/AWS/TH.hs:37:10: warning: [-Wmissing-methods]
    • No explicit implementation for
        ‘liftTyped’
    • In the instance declaration for ‘Lift Day’
   |
37 | instance Lift Day where
   |          ^^^^^^^^
```

--- a/test/src/Test/AWS/Fixture.hs
+++ b/test/src/Test/AWS/Fixture.hs
@@ -25,7 +25,6 @@ import qualified Data.Conduit.Binary as Conduit
 import qualified Data.List as List
 import qualified Data.Ord as Ord
 import qualified Data.Text.Encoding as Text
-import Data.Time (UTCTime (..))
 import qualified Data.Yaml as YAML
 import Network.AWS.Core
 import Network.AWS.Prelude

--- a/test/src/Test/AWS/TH.hs
+++ b/test/src/Test/AWS/TH.hs
@@ -1,3 +1,6 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveLift #-}
+{-# LANGUAGE StandaloneDeriving #-}
 {-# LANGUAGE TemplateHaskell #-}
 {-# OPTIONS_GHC -fno-warn-orphans #-}
 
@@ -10,7 +13,7 @@
 -- Portability : non-portable (GHC extensions)
 module Test.AWS.TH where
 
-import Data.Time
+import Data.Time (Day (..), DiffTime, UTCTime (..))
 import Language.Haskell.TH
 import Language.Haskell.TH.Syntax
 import Network.AWS.Core
@@ -22,17 +25,16 @@ mkTime x =
     Left e -> error (show e)
     Right t -> [|view _Time t|]
 
-instance Lift (Time a) where
-  lift (Time x) = AppE (ConE (mkName "Time")) <$> lift x
+deriving instance Lift (Time a)
 
-instance Lift UTCTime where
-  lift (UTCTime x y) = do
-    x' <- lift x
-    y' <- lift y
-    return $ AppE (AppE (ConE (mkName "UTCTime")) x') y'
+deriving instance Lift UTCTime
 
+deriving instance Lift Day
+
+-- DiffTime's constructor is not exported, so use a manual instance
 instance Lift DiffTime where
+#if MIN_VERSION_template_haskell(2,16,0)
+  liftTyped x = [||toEnum $$(liftTyped (fromEnum x))||]
+#else
   lift x = [|toEnum $(lift (fromEnum x))|]
-
-instance Lift Day where
-  lift x = [|toEnum $(lift (fromEnum x))|]
+#endif

